### ISSUE TITLE
acceptance: schema change nightly failing

### ISSUE THREAD

**Issue by cuongdo** [cockroachdb/cockroach#17922]
The last 3 runs have failed. [The latest run](https://teamcity.cockroachdb.com/viewLog.html?buildId=334093&tab=buildResultsDiv&buildTypeId=Cockroach_Nightlies_SchemaChangeTest3to510g) has a lot of errors in the `cockroach` logs.

Note that the same test without the schema change has not had the same issue.

Here are the earlier logs showing issues:

```
E170823 06:17:29.936034 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:29.993222 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:30.094270 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
I170823 06:17:30.170125 99 storage/replica_proposal.go:453  [n1,s1,r54/1:/Table/51/1/15{45226â€¦-82165â€¦}] new range lease repl=(n1,s1):1 start=1464887669.045577162,1 epo=1 pro=1503469050.164138313,0 following repl=(n1,s1):1 start=1464887669.045577162,1 exp=1464892575.180085507,0
E170823 06:17:30.276232 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:30.668274 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:31.494723 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:32.525190 50 sqlmigrations/migrations.go:224  [n1] failed attempt to acquire migration lease: lease /System/"system-version/lease" is not available until at least 1503469106.521585221,0
E170823 06:17:33.485017 24 sql/jobs/registry.go:143  error while adopting jobs: relation "system.jobs" does not exist
```

then lots of:

```
I170823 06:17:35.589290 140 storage/replicate_queue.go:203  [replicate,n1,s1,r171/1:/Table/51/1/4{072711â€¦-106331â€¦}] no rebalance target found, not enqueuing
```

and

```
I170823 06:17:53.647708 140 storage/allocator.go:241  [replicate,n1,s1,r21/1:/Table/51/1/1{697144â€¦-730631â€¦}] unable to take action - live replicas [] don't meet quorum of 2
```

cc @a-robinson in case he has ideas

**Comment by a-robinson** [comment]
The errors of the form `error restarting node 2: cat cockroachdb-url: Process exited with status 1` indicate that the node didn't start to the point where it had written its listening URL to the url file. The question is why the nodes are struggling to start. Given that the test appears to fail pretty consistently now, I'd say the most suspicious-looking PRs between the previous green run and the start of the failures are https://github.com/cockroachdb/cockroach/pull/17624 and https://github.com/cockroachdb/cockroach/pull/17810

**Comment by cuongdo** [comment]
For posterity, it looks like the last schema change test that had stats-based rebalancing enabled timed out after 24 hours:

https://teamcity.cockroachdb.com/viewLog.html?buildId=335827&buildTypeId=Cockroach_Nightlies_SchemaChangeTest3to510g&tab=buildLog

**Comment by vivekmenezes** [comment]
@cuongdo I'm planning on looking at this after #17989 is fixed. Does that make sense?

**Comment by a-robinson** [comment]
It doesn't make much sense given that this test has been failing more consistently since a more well-defined day, but it does make sense given that they're potentially hitting the same issue. Feel free to wait.

**Comment by vivekmenezes** [comment]
It's failing in a number of places. Let me at least run it by hand and see
if it works

**Comment by cuongdo** [comment]
This is now failing for a new reason:

```
allocator_test.go:221: hit schema change error: pq: relation "datablocks.blocks" does not exist, for ALTER TABLE datablocks.blocks ADD COLUMN newcol DECIMAL DEFAULT (DECIMAL '1.4'), in 379.084447ms
```

The underlying data has moved from `block_writer`'s schema to `tpch`'s schema. So, a new schema change that's appropriate for `tpch` is now needed.

**Comment by jordanlewis** [comment]
@arjunravinarayan any advice here? What's a TPCH table that has a lot of rows?

**Comment by cuongdo** [comment]
That would be `lineitem.tpch`. For the given scale factor (10), that should have ~30 million rows.

**Comment by arjunravinarayan** [comment]
@jordanlewis do you have any further requirements for what sort of schema changes you want to run? If you just want a bunch of schema change jobs, all the `CREATE INDEX` statements in `loadgen/tpch/ddls.go` are perfect candidates, as there's a bunch of them, and they're all ready made and available.

**Comment by vivekmenezes** [comment]
@jordanlewis are you looking at fixing this?

**Comment by vivekmenezes** [comment]
BTW I see the block_writer and photos still running in the test so I'm confused as to why it is failing unable to find the schema for those tests.

I think this test should reflect both the block_writer and photos schema changes (tests schema change in the midst of updates) and the TPCH schema changes (tests schema change over a large table).

**Comment by cuongdo** [comment]
I think this is a race between the schema change and `block_writer` creating its schema.

**Comment by vivekmenezes** [comment]
This has been fixed by #18444 and #18369
